### ISSUE TITLE
openshift_hosted_facts task failed merge two dicts

### ISSUE THREAD

**Issue by mrsiano** [openshift/openshift-ansible#4985]
#### Description
when using config playbook, ansible-playbook -i inventory/byo/hosts playbooks/byo/config.yml

I've got the following error:

##### Version
ansible 2.3.1.0

* The output of `git describe`
openshift-ansible-3.7.1-1-76-g1765ce2

Place the output between the code block below:

task path: /root/openshift-ansible/roles/openshift_hosted_facts/tasks/main.yml:9
Thursday 03 August 2017  07:37:22 -0400 (0:00:00.072)       0:01:28.574 ******* 
fatal: [70.70.70.1]: FAILED! => {
    "failed": true
}

MSG:

|failed expects to merge two dicts

##### Steps To Reproduce
1. [just run ansible-playbook]

##### Expected Results
deployment should pass successfully. 

##### Observed Results
Failure summary:

  1. Host:     ...
     Play:     Configure masters
     Task:     openshift_hosted_facts : Set hosted facts
     Message:  |failed expects to merge two dicts

**Comment by mrsiano** [comment]
seems like the problem related to oo_merge_hostvars, the last change was by commit 'f0c412a9ab50a9030d51f83c459a66451a7c344a' 
@smarterclayton could your review this

**Comment by sdodson** [comment]
@mrsiano can you please provide your inventory?

**Comment by mrsiano** [comment]
@sdodson sure thing 

[OSEv3:children]
masters
nodes

[OSEv3:vars]
ansible_ssh_user=root
ansible_become=false
qe_openshift_kerberos_user=emarcian@redhat.com
qe_openshift_token=somehashkey
aos_repo=someprivaterepo
debug_level=2
deployment_type=openshift-enterprise
openshift_install_examples=true
oreg_url=registry.ops.openshift.com/openshift3/ose-${component}:${version}
openshift_master_identity_providers=[{'name': 'allow_all', 'login': 'true', 'challenge': 'true', 'kind': 'AllowAllPasswordIdentityProvider'}]
os_sdn_network_plugin_name='redhat/openshift-ovs-multitenant'
openshift_node_kubelet_args={'max-pods': ['200'], 'image-gc-high-threshold': ['90'], 'image-gc-low-threshold': ['80'] }
# Disable cockpit
osm_use_cockpit=false
openshift_disable_check=disk_availability,docker_storage,docker_image_availability,package_availability,package_version
openshift_ip=70.70.70.1
openshift_host_ip=70.70.70.1

[masters]
70.70.70.1
[nodes]
70.70.70.1 openshift_schedulable=True openshift_node_labels="{'region':'infra','zone':'default'}"
70.70.70.2 openshift_node_labels={'region':'infra','zone':'default'}

**Comment by abutcher** [comment]
@mrsiano I wasn't able to reproduce with this inventory so it might have something to do with the scrubbed variables. Could you send the full inventory excluding tokens to my RH mail?

`openshift_ip` is set globally and should be set at the host var level.

```
[masters]
70.70.70.1 openshift_ip=70.70.70.1 openshift_public_ip=70.70.70.1

[nodes]
70.70.70.1 openshift_schedulable=True openshift_node_labels="{'region':'infra','zone':'default'}" openshift_ip=70.70.70.1 openshift_public_ip=70.70.70.1
70.70.70.2 openshift_node_labels={'region':'infra','zone':'default'} openshift_ip=70.70.70.2 openshift_public_ip=70.70.70.2
```

**Comment by abutcher** [comment]
@mrsiano were you able to work around? I wasn't able to reproduce with the full inventory.

**Comment by mrsiano** [comment]
@abutcher yep, I just comment out those lines.
I'll try to refresh and pull again the project, will let you know next..

**Comment by Pectojin** [comment]
I've replicated the issue on my setup. I get the same error when running under ansible 2.6.0.

Inventory versions:

    openshift_release=v3.7
    openshift_pkg_version=-3.7.1
    openshift_image_tag=v3.7.1

And downgrading to ansible 2.4.2 indeed lets me continue without this error.

**Comment by alanbchristie** [comment]
I also _tripped up_ moving to Ansible `2.6.1` on the OpenShift 3.6 release. I'm using the **openshift-ansible-3.6.173.0.120-1** tag and am OK using Ansible `2.5.3` but encountered the following with Ansible 2.6.1: -

```
TASK [openshift_hosted_facts : Set hosted facts] ****************************************************************
	Sunday 08 July 2018  09:35:31 +0100 (0:00:00.074)       0:03:48.518 *********** 
	fatal: [18.195.236.210]: FAILED! => {"msg": "|failed expects to merge two dicts"}
```

**Comment by sdodson** [comment]
/assign @mtnbikenc @vrutkovs 
2.6 items to fix according to most recent comment

**Comment by vrutkovs** [comment]
I tried to update this to support both 2.4 (`dict`) and 2.6 (`ansible.vars.hostvars.HostVarsVars`), but it would require a complex code for 3.7 and lower to support this

**Comment by slaterx** [comment]
NOT reproducible under ansible 2.4.6 and OpenShift origin 3.7.

**Comment by vrutkovs** [comment]
Fixed by #9125
